pallets/pools/src/lib.rs

```diff
@@ -1524,20 +1607,57 @@ pub mod pallet {
 
 					pool.tranches.combine_with_mut_residual_top(
 						tranches.into_iter(),
-						|tranche, (new_tranche_type, seniority)| {
+						|tranche, tranche_update| {
 							// Update debt of the tranche such that the interest is accrued until now with the previous interest rate
 							tranche.accrue(now)?;
 
-							tranche.tranche_type = new_tranche_type.clone();
+							tranche.tranche_type = tranche_update.tranche_type.clone();
 
-							if let Some(new_seniority) = seniority {
+							if let Some(new_seniority) = tranche_update.seniority {
 								tranche.seniority = new_seniority.clone();
 							}
+
 							Ok(())
 						},
 					)?;
 				}
 
+				// Checking if we have a new tranche?
+
+				//
+				// The case when Metadata AND the tranche changed, we don't allow for an or. Both have to be changed (for now)
+				//
+				if let Change::NewValue(metadata) = &changes.tranche_metadata {
```

Comment:
That's my attempt to update metadata for each tranche. WhatI could not figure out yet how these two are associated. I assume we need the `tranche_id`, could not figure out yet how to get it, since the `tranche` from the parameters don't have the `tranche_id`.